File: <3>菜单/搜索栏类型的菜单/Menu/ViewController.m
//
//  ViewController.m
//  Menu
//
//  Created by sangcixiang on 16/2/27.
//  Copyright © 2016年 sangcixiang. All rights reserved.
//

#import "ViewController.h"
#import "CSNavigationBarMenu.h"
@interface ViewController ()
- (IBAction)search:(UIBarButtonItem *)sender;

@property(nonatomic,strong)CSNavigationBarMenu * menu;
@property(nonatomic,assign)BOOL isShow;
@end

@implementation ViewController
-(CSNavigationBarMenu *)menu
{
    if (!_menu)
    {                                                       /** 设置菜单的位置及大小  */
        _menu = [[CSNavigationBarMenu alloc]initWithOrigin:CGPointMake(5,50) width:150];
    }
    return _menu;
}
- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    
}
- (IBAction)search:(UIBarButtonItem *)sender {
    if (self.isShow) {
        [self.menu dismiss];
        self.isShow = NO;
    }else if(!self.isShow){
        [self.menu show];
        self.isShow = YES;
    }
}











@end

File: <7>动画/随按钮的拖拽而变化头像/MCCircleView/ViewController.swift
//
//  ViewController.swift
//  MCCircleView
//
//  Created by 马超 on 16/5/16.
//  Copyright © 2016年 qq 714080794. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var sliderView: UISlider!
    
    @IBOutlet weak var circleView: CircleView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        
    }

  

    @IBAction func sliderChanged(sender: UISlider) {
        
        self.circleView.circleLayer.updataWithProgress(CGFloat(sender.value))
        
    }

}

